tests/components/forked_daapd/test_browse_media.py

```diff
@@ -0,0 +1,294 @@
+                            "path": "/music/srv/input.pipe",
+                            "uri": "library:track:1",
+                            "artwork_url": "/artwork/item/1",
+                        }
+                    ],
+                    "total": 1,
+                    "offset": 0,
+                    "limit": -1,
+                },
+                "playlists": {
+                    "items": [
+                        {
+                            "id": 8,
+                            "name": "radio",
+                            "path": "/music/srv/radio.m3u",
+                            "smart_playlist": True,
+                            "uri": "library:playlist:8",
+                        }
+                    ],
+                    "total": 1,
+                    "offset": 0,
+                    "limit": -1,
+                },
+            }
+        ] + 4 * [
+            {"directories": [], "tracks": {"items": []}, "playlists": {"items": []}}
+        ]
+        mock_api.return_value.get_albums.return_value = [
+            {
+                "id": "8009851123233197743",
+                "name": "Add Violence",
+                "name_sort": "Add Violence",
+                "artist": "Nine Inch Nails",
+                "artist_id": "32561671101664759",
+                "track_count": 5,
+                "length_ms": 1634961,
+                "uri": "library:album:8009851123233197743",
+            },
+        ]
+        mock_api.return_value.get_artists.return_value = [
+            {
+                "id": "3815427709949443149",
+                "name": "ABAY",
+                "name_sort": "ABAY",
+                "album_count": 1,
+                "track_count": 10,
+                "length_ms": 2951554,
+                "uri": "library:artist:3815427709949443149",
+            },
+        ]
+        mock_api.return_value.get_genres.return_value = [
+            {"name": "Classical"},
+            {"name": "Drum & Bass"},
+            {"name": "Pop"},
+            {"name": "Rock/Pop"},
+            {"name": "'90s Alternative"},
+        ]
+        mock_api.return_value.get_playlists.return_value = [
+            {
+                "id": 1,
+                "name": "radio",
+                "path": "/music/srv/radio.m3u",
+                "smart_playlist": False,
+                "uri": "library:playlist:1",
+            },
+        ]
+
+        # Request playlist through WebSocket
+        client = await hass_ws_client(hass)
+        await client.send_json(
+            {
+                "id": 1,
+                "type": "media_player/browse_media",
+                "entity_id": TEST_MASTER_ENTITY_NAME,
+            }
+        )
+        msg = await client.receive_json()
+        # Assert WebSocket response
+        assert msg["id"] == 1
+        assert msg["type"] == TYPE_RESULT
+        assert msg["success"]
+
+        msg_id = 2
+
+        async def browse_children(children):
```

Comment:
We can inline this function. It's only called once.

Reply:
It's a recursive function - how to inline it?

Reply:
I didn't see that it's called twice, not once.